tensorflow/python/keras/utils/vis_utils.py

```diff
@@ -63,6 +64,34 @@ def add_edge(dot, src, dst):
     dot.add_edge(pydot.Edge(src, dst))
 
 
+def is_regex_match(pat, text):
+  check = re.match(pat, text)
+  return False if check is None else True
+
+
+def get_layer_index_bound_by_layer_name(model, layer_names):
+  '''Return specific range of layers to plot, mainly for sub-graph plot models.
+  Args:
+    model: tf.keras.Model 
+    layer_names: unique name of layer of the model, type(str)
+
+  Returns:
+    retun the index value of layer based on its unique name (layer_names)
+  '''
+  lower_index = []
+  upper_index = []
+  for idx, layer in enumerate(model.layers):
+    if is_regex_match(layer_names[0], layer.name):
+      lower_index.append(idx)
+    if is_regex_match(layer_names[1], layer.name):
+      upper_index.append(idx)
+  if len(lower_index) == 0 or len(upper_index) == 0:
+    raise ValueError("Passed layer_range does not match to model layers")
+  if min(lower_index) > max(upper_index):
+    return [max(upper_index), min(lower_index)]
```

Comment:
Should we return min(upper_index) and max(lower_index) here?

Reply:
done